vertical-pod-autoscaler/pkg/recommender/routines/recommender.go

```diff
@@ -122,12 +155,16 @@ func (r *recommender) UpdateVPAs() {
 		}
 		cnt.Add(vpa)
 
-		_, err := vpa_utils.UpdateVpaStatusIfNeeded(
+		_, err = vpa_utils.UpdateVpaStatusIfNeeded(
 			r.vpaClient.VerticalPodAutoscalers(vpa.ID.Namespace), vpa.ID.VpaName, vpa.AsStatus(), &observedVpa.Status)
 		if err != nil {
 			klog.Errorf(
 				"Cannot update VPA %v/%v object. Reason: %+v", vpa.ID.Namespace, vpa.ID.VpaName, err)
 		}
+		err = vpa_utils.PatchVpaAnnotations(r.vpaClient.VerticalPodAutoscalers(vpa.ID.Namespace), vpa.ID.VpaName, vpa.Annotations)
```

Comment:
Should the vpa annotations patch only occur if the annotations have changed? (similar to the status?) I tried to implement this at first but it seemed like the annotations didnt update in that case (is this an issue with the observed VPA having the same annotations?)

Reply:
When the recommendation (status) is changed then annotations have to be updated as well. When the recommendation (status) has not changed then we should update the annotations only once in a while (ideally a configuration parameter) e.g. not more often than once per hour. For example, the code might check if last update timestamp is older than 1h and if it is, patch the annotations (if new annotations are different than the old annotations). The goal is to minimize unnecessary load on API server.